src/libraries/Microsoft.Bcl.Cryptography/src/System/Security/Cryptography/NetStandardShims.cs

```diff
@@ -21,6 +35,96 @@ internal static unsafe int GetBytes(this Encoding encoding, ReadOnlySpan<char> s
                 return encoding.GetBytes(pStr, str.Length, pDestination, destination.Length);
             }
         }
+
+        internal static void ReadExactly(this System.IO.Stream stream, Span<byte> buffer) =>
+            ReadAtLeast(stream, buffer, buffer.Length, throwOnEndOfStream: true);
```

Comment:
Is delegating this to `ReadAtLeast` correct? If `buffer` is two bytes in length, you are saying it is okay to read three bytes, at which the stream has probably read more than you are accounting for and you threw away one byte.

Reply:
`ReadAtLeast` can't read more than the size of the buffer, so it's really `ReadRange(minimumBytes, buffer.Length)`.  If `minimumBytes == buffer.Length` the upper and lower bounds are the same, so it's the same as `ReadExactly`.